server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -283,6 +286,35 @@ public boolean setExpiration(String name, Instant expiration) {
     return Optional.empty();
   }
 
+  /**
+   * @param names of secrets series to look up secrets by.
+   * @return Secrets matching input parameters.
+   */
+  public List<SecretSeriesAndContent> getSecretsByName(List<String> names) {
```

Comment:
I've seen in other places we had API such that it would return `Optional<ImmutableList>`. I'm not sure if that's better, but there's a question about overall API consistency here.

Reply:
There's a couple of things I'd like to make more consistent, especially around testing. We should earmark brushing up this code for another hackweek.